test/known_problems/should_fail/map_expr.erl

```diff
@@ -0,0 +1,8 @@
+-module(map_expr).
+
+-export([foo/0]).
+
+-spec foo() -> ok.
+foo() ->
+    A = #{ok => 42},
+    A.
```

Comment:
I think this would fail with `-gradualizer(infer)` but maybe it should not fail without infer, since in general literals are considered untyped. (Then, we may consider changing this, but that's another story.)

Reply:
It doesn't fail with infer either 😢. Should I move this to `known_problems/should_fail_infer` (and add support for that folder)?

Reply:
Just add `-gradualizer(infer).` to the file, perhaps. No need for separate folder IMO.